include/bluetooth/mesh/vnd/dm_cli.h

```diff
@@ -0,0 +1,225 @@
+ * @param[in] ctx Message context, or NULL to use the configured publish
+ * parameters.
+ * @param[in] set New configuration parameters, or NULL to get the current parameters.
+ * @param[out] rsp Status response buffer, or NULL to keep from blocking.
+ *
+ * @retval 0 Successfully sent the message and populated the @p rsp buffer.
+ * @retval -EALREADY A blocking request is already in progress.
+ * @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ * not configured.
+ * @retval -EAGAIN The device has not been provisioned.
+ * @retval -ETIMEDOUT The request timed out without a response.
+ * @retval -EBADMSG The supplied configuration parameters are invalid.
+ */
+int bt_mesh_dm_cli_config(struct bt_mesh_dm_cli *cli,
+			  struct bt_mesh_msg_ctx *ctx,
+			  const struct bt_mesh_dm_cfg *set,
+			  struct bt_mesh_dm_cli_cfg_status *rsp);
+
+/** @brief Start a distance measurement on a Distance Measurement server.
+ *
+ * This call is blocking if the @p rsp buffer is non-NULL. Otherwise, this
+ * function will return, and the response will be passed to the
+ * @ref bt_mesh_dm_cli_handlers::result_handler callback.
+ *
+ * @param[in] cli Client model to send on.
+ * @param[in] ctx Message context, or NULL to use the configured publish
+ * parameters.
+ * @param[in] start Distance measurement Start parameters.
+ * @param[out] rsp Status response buffer, or NULL to keep from blocking.
+ *
+ * @retval 0 Successfully sent the message and populated the @p rsp buffer.
+ * @retval -EALREADY A blocking request is already in progress.
+ * @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ * not configured.
+ * @retval -EAGAIN The device has not been provisioned.
+ * @retval -ETIMEDOUT The request timed out without a response.
+ * @retval -EBADMSG The supplied configuration parameters are invalid.
+ */
+int bt_mesh_dm_cli_measurement_start(struct bt_mesh_dm_cli *cli, struct bt_mesh_msg_ctx *ctx,
+				    const struct bt_mesh_dm_cli_start *start,
+				    struct bt_mesh_dm_cli_result *rsp);
+
+/** @brief Get measurement results from a Distance Measurement server.
+ *
+ * This call is blocking if the @p rsp buffer is non-NULL. Otherwise, this
+ * function will return, and the response will be passed to the
+ * @ref bt_mesh_dm_cli_handlers::result_handler callback.
+ *
+ * @param[in] cli Client model to send on.
+ * @param[in] ctx Message context, or NULL to use the configured publish
+ * parameters.
+ * @param[in] entry_cnt Number of entries the user wish to receive.
+ * @param[out] rsp Status response buffer, or NULL to keep from blocking.
+ *
+ * @retval 0 Successfully sent the message and populated the @p rsp buffer.
+ * @retval -EALREADY A blocking request is already in progress.
+ * @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ * not configured.
+ * @retval -EAGAIN The device has not been provisioned.
+ * @retval -ETIMEDOUT The request timed out without a response.
+ */
```

Comment:
This can now return EBADMSG if you request too many entries

Reply:
Backlog item.